dask/array/numpy_compat.py

```diff
@@ -359,3 +361,27 @@ def nancumprod(a, axis=None, dtype=None, out=None):
         """
         a, mask = _replace_nan(a, 1)
         return np.cumprod(a, axis=axis, dtype=dtype, out=out)
+
+
+def _make_sliced_dtype(dtype, index):
+    if LooseVersion(np.__version__) >= LooseVersion("1.14.0"):
+        return _make_sliced_dtype_new(dtype, index)
+    else:
+        return _make_sliced_dtype_old(dtype, index)
+
+
+def _make_sliced_dtype_new(dtype, index):
+    # For https://github.com/numpy/numpy/pull/6053
+    # TODO: handle either positional or named indexing
```

Comment:
Oh, this thankfully won't have to deal with positional indexing, since that's handled elsewhere in `__getitem__`.